python-spec/src/somacore/collection.py

```diff
@@ -83,43 +87,55 @@ def add_new_collection(
         as in :meth:`create`. By default, this will create a basic collection::
 
             # Create a child Measurement object at the key "density"
             # with default settings.
-            #
-            # This creates both the backing Measurement collection, as well as
-            # the necessary sub-elements to have a complete Measurement.
             density = the_collection.add_new_collection("density", somacore.Measurement)
 
-            # This will create a basic Collection as a child.
-            sub_child = density.add_new_collection("sub_child")
-
-        In situations where relative URIs are supported, the default URI of the
-        child should be a sub-path of the parent URI. The final component
-        should be a path-sanitized version of the new key. For instance, adding
-        sub-element ``data`` to a Collection located at ``file:///path/to/coll``
-        would have a default path of ``file:///path/to/coll/data``. For
-        non–URI-safe keys, no specific path-sanitization method is required,
-        but as an example, a child named ``non/path?safe`` could use
-        ``file:///path/to/coll/non_path_safe`` as its full URI.
-
-        When a specific child URI is specified, that exact URI should be used
-        (whether relative or absolute).
-
-        :param key: The key that this child should live at (i.e., it will be
-            accessed via ``the_collection[key]``).
-        :param kind: The type of child that should be added.
-        :param uri: If provided, overrides the default URI that would be used
-            to create this object. This may be absolute or relative.
-        :param platform_config: Platform-specific configuration options used
-            when creating the child.
+            # This will create a basic Collection as a child at the location
+            # storage://authority/path/to/sub-child.
+            sub_child = density.add_new_collection(
+                "sub_child", uri="storage://authority/path/to/sub-child")
+
+        The URI provided may be absolute or relative. If a child URI is not
+        provided, the collection should construct a default child URI based
+        on the key of the new entry, making a relative URI (when possible)::
+
+            # coll URI is "file:///path/to/coll"
+            coll.add_new_collection("new child!")
+            # The URI of the child collection might be:
+            # "file:///path/to/coll/new_child"
+
+            # flat_ns_coll URI is "flat://authority/key" in a flat namespace
+            # where relative paths are unsupported.
+            flat_ns_coll.add_new_collection("flat child")
+            # The URI of the child collection might be:
+            # "flat://authority/flat_child"
+
+        The way the URI is constructed is left unspecified so that an
+        implementation can create a URI based on its own needs. Users should
+        directly get the URI of the new child if they need it; they should never
```

Comment:
worth providing an example of how to get the URI?

Reply:
Added.